.github/scripts/strategy-matrix/generate.py

```diff
@@ -238,20 +240,156 @@ def generate_strategy_matrix(all: bool, config: Config) -> list:
+        extra_warning_flags += " -Wno-stringop-overflow"
+        # Disable mold, gold and lld linkers.
+        # Use default linker (bfd/ld) which is more lenient with mixed code models
+        cmake_args += " -Duse_mold=OFF -Duse_gold=OFF -Duse_lld=OFF"
+        # Add linker flags for Sanitizers
+        linker_flags += f" -DCMAKE_EXE_LINKER_FLAGS='{linker_relocation_flags} -fsanitize=address,{sanitizers_flags}'"
+        linker_flags += f" -DCMAKE_SHARED_LINKER_FLAGS='{linker_relocation_flags} -fsanitize=address,{sanitizers_flags}'"
+    elif os["compiler_name"] == "clang":
+        # Note: We use $GITHUB_WORKSPACE environment variable which will be expanded by the shell
+        # before CMake processes it. This ensures the compiler receives an absolute path.
+        # CMAKE_SOURCE_DIR won't work here because it's inside CMAKE_CXX_FLAGS string.
+        # GCC doesn't support ignorelist.
+        cxx_flags += " -fsanitize-ignorelist=$GITHUB_WORKSPACE/sanitizers/suppressions/sanitizer-ignorelist.txt"
+        sanitizers_flags = (
+            f"{sanitizers_flags},signed-integer-overflow,unsigned-integer-overflow"
+        )
+        linker_flags += (
+            f" -DCMAKE_EXE_LINKER_FLAGS='-fsanitize=address,{sanitizers_flags}'"
+        )
+        linker_flags += (
+            f" -DCMAKE_SHARED_LINKER_FLAGS='-fsanitize=address,{sanitizers_flags}'"
+        )
+
+    # Sanitizers recommend minimum of -O1 for reasonable performance
+    cxx_flags += " -O1"
+
+    # First create config for asan
+    cmake_args_flags = f'{cmake_args} -DCMAKE_CXX_FLAGS="-fsanitize=address,{sanitizers_flags} -fno-omit-frame-pointer {cxx_flags} {extra_warning_flags}" {linker_flags}'
+
+    # Add config with asan
+    configurations.append(
+        {
+            "config_name": config_name + "_asan",
+            "cmake_args": cmake_args_flags,
+            "cmake_target": cmake_target,
+            "build_only": build_only,
+            "build_type": build_type,
+            "os": os,
+            "architecture": architecture,
+            "sanitizers": "Address",
+        }
+    )
+    # Since TSAN runs are crashing with seg faults(could be compatibility issues with latest compilers)
+    # We deactivate it for now. But I would keep the code, since it took some effort to find the correct set of config needed to run this.
+    # This will be useful when we decide to activate it again in future.
+    activateTSAN = False
+    if activateTSAN:
+        linker_flags = ""
+        # Update configs for tsan
+        # gcc doesn't supports atomic_thread_fence with tsan. Suppress warnings.
+        # Also tsan doesn't work well with mcmode=large and bfd linker
+        if os["compiler_name"] == "gcc":
+            extra_warning_flags += " -Wno-tsan"
+            cxx_flags = cxx_flags.replace("-mcmodel=large", "-mcmodel=medium")
+            linker_relocation_flags = linker_relocation_flags.replace(
+                "-mcmodel=large", "-mcmodel=medium"
+            )
```

Comment:
It's quite difficult to understand what end up being in these flags, especially because of replace, can we rewrite this in a more clear way?